yt_dlp/extractor/niconicochannelplus.py

```diff
@@ -0,0 +1,428 @@
+
+        release_timestamp_str = data_json.get('live_scheduled_start_at')
+
+        formats = []
+
+        if live_status == 'is_upcoming':
+            if release_timestamp_str:
+                msg = f'This live event will begin at {release_timestamp_str} UTC'
+            else:
+                msg = 'This event has not started yet'
+            self.raise_no_formats(msg, expected=True, video_id=content_code)
+        else:
+            formats = self._extract_m3u8_formats(
+                # "authenticated_url" is a format string contains "{session_id}".
+                m3u8_url=data_json['video_stream']['authenticated_url'].format(session_id=session_id),
+                video_id=content_code)
+
+        return {
+            'id': content_code,
+            'formats': formats,
+            '_format_sort_fields': ('tbr', 'vcodec', 'acodec'),
+            'channel': self._get_channel_base_info(fanclub_site_id).get('fanclub_site_name'),
+            'channel_id': channel_id,
+            'channel_url': f'{self._WEBPAGE_BASE_URL}/{channel_id}',
+            'age_limit': traverse_obj(self._get_channel_user_info(fanclub_site_id), ('content_provider', 'age_limit')),
+            'live_status': live_status,
+            'release_timestamp': unified_timestamp(release_timestamp_str),
+            **traverse_obj(data_json, {
+                'title': ('title', {str}),
+                'thumbnail': ('thumbnail_url', {url_or_none}),
+                'description': ('description', {str}),
+                'timestamp': ('released_at', {unified_timestamp}),
+                'duration': ('active_video_filename', 'length', {int_or_none}),
+                'comment_count': ('video_aggregate_info', 'number_of_comments', {int_or_none}),
+                'view_count': ('video_aggregate_info', 'total_views', {int_or_none}),
+                'tags': ('video_tags', ..., 'tag', {str}),
+            }),
+            '__post_extractor': self.extract_comments(
+                content_code=content_code,
+                comment_group_id=traverse_obj(data_json, ('video_comment_setting', 'comment_group_id'))),
+        }
+
+    def _get_comments(self, content_code, comment_group_id):
+        item_id = f'{content_code}/comments'
+
+        if not comment_group_id:
+            return None
+
+        comment_access_token = self._call_api(
+            f'video_pages/{content_code}/comments_user_token', item_id,
+            note='Getting comment token', errnote='Unable to get comment token',
+        )['data']['access_token']
+
+        comment_list = self._download_json(
+            'https://comm-api.sheeta.com/messages.history', video_id=item_id,
+            note='Fetching comments', errnote='Unable to fetch comments',
+            headers={'Content-Type': 'application/json'},
+            query={
+                'sort_direction': 'asc',
+                'limit': traverse_obj(self._configuration_arg('max_comments', [120]), (0, )),
```

Comment:
user input will be string, we need to make it int
```suggestion
                'limit': int_or_none(self._configuration_arg('max_comments', [''])[0]) or 120,
```